File: csharp/CsSearch/CsSearchTests/SearchFileTests.cs
using NUnit.Framework;
using CsSearchLib;

namespace CsSearchTests
{
	[TestFixture]
	class SearchFileTests
	{
		private const string CsSearchPath = "~/src/xsearch/csharp/CsSearch/CsSearch";
		private const string WinCsSearchPath = @"C:\src\git\xsearch\csharp\CsSearch\CsSearch";

		[Test]
		public void SearchFile_ToString_EqualsExpected()
		{
			var searchFile = new SearchFile(CsSearchPath, "Searcher.cs", FileType.Text);
			Assert.AreEqual(CsSearchPath + "/Searcher.cs", searchFile.ToString());
		}

		[Test]
		public void SearchFileTrailingSlash_ToString_EqualsExpected()
		{
			var searchFile = new SearchFile(CsSearchPath + "/", "Searcher.cs", FileType.Text);
			Assert.AreEqual(CsSearchPath + "/Searcher.cs", searchFile.ToString());
		}

		[Test]
		public void SearchFileBackSlashes_ToString_EqualsExpected()
		{
			var searchFile = new SearchFile(WinCsSearchPath, "Searcher.cs", FileType.Text);
			Assert.AreEqual(WinCsSearchPath + @"\Searcher.cs", searchFile.ToString());
		}

		[Test]
		public void SearchFileBackSlashesTrailingSlash_ToString_EqualsExpected()
		{
			var searchFile = new SearchFile(WinCsSearchPath + @"\", "Searcher.cs", FileType.Text);
			Assert.AreEqual(WinCsSearchPath + @"\Searcher.cs", searchFile.ToString());
		}
	}
}

File: fsharp/FsSearch/FsSearchTests/SearchFileTests.fs
namespace FsSearchTests

open System.IO
open NUnit.Framework
open FsSearch

[<TestFixture>]
type SearchFileTests () =

    member this.CsSearchPath = "~/src/xsearch/csharp/CsSearch/CsSearch"
    member this.WinCsSearchPath = @"C:\src\xsearch\csharp\CsSearch\CsSearch"

    [<SetUp>]
    member this.Setup () =
        ()

    [<Test>]
    member this.SearchFile_ToString_EqualsExpected () =
        let searchFile = SearchFile.Create (FileInfo(FileUtil.JoinPath this.CsSearchPath "Searcher.cs")) FileType.Code
        Assert.AreEqual(this.CsSearchPath + "/Searcher.cs", SearchFile.ToString(searchFile))
        ()

    [<Test>]
    member this.SearchFileTrailingSlash_ToString_EqualsExpected () =
        let searchFile = SearchFile.Create (FileInfo(FileUtil.JoinPath this.CsSearchPath "Searcher.cs")) FileType.Code
        Assert.AreEqual(this.CsSearchPath + "/Searcher.cs", SearchFile.ToString(searchFile))
        ()

    [<Test>]
    member this.SearchFileBackSlashes_ToString_EqualsExpected () =
        let searchFile = SearchFile.Create (FileInfo(FileUtil.JoinPath this.WinCsSearchPath "Searcher.cs")) FileType.Code
        Assert.AreEqual(this.WinCsSearchPath + @"\Searcher.cs", SearchFile.ToString(searchFile))
        ()

    [<Test>]
    member this.SearchFileBackSlashesTrailingSlash_ToString_EqualsExpected () =
        let searchFile = SearchFile.Create (FileInfo(FileUtil.JoinPath this.WinCsSearchPath "Searcher.cs")) FileType.Code
        Assert.AreEqual(this.WinCsSearchPath + @"\Searcher.cs", SearchFile.ToString(searchFile))
        ()
